x-pack/plugins/security_solution/public/cases/components/connectors/case/fields.tsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/* eslint-disable @kbn/eslint/no-restricted-paths */
+
+import React, { useCallback, useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { EuiCallOut } from '@elastic/eui';
+
+import { ActionParamsProps } from '../../../../../../triggers_actions_ui/public/types';
+import { CommentType } from '../../../../../../case/common/api';
+
+import { CaseActionParams } from './types';
+import { ExistingCase } from './existing_case';
+
+import * as i18n from './translations';
+
+const Container = styled.div`
+  ${({ theme }) => `
+    margin-top: ${theme.eui?.euiSize ?? '16px'};
+  `}
+`;
+
+const CaseParamsFields: React.FunctionComponent<ActionParamsProps<CaseActionParams>> = ({
+  actionParams,
+  editAction,
+  index,
+  errors,
+  messageVariables,
+  actionConnector,
+  http,
+  toastNotifications,
+}) => {
+  const {
+    caseId = null,
+    comment = {
+      type: CommentType.alert,
+      alertId: '{{context.rule.id}}',
+      index: '{{context.rule.output_index}}',
+    },
+  } = actionParams.subActionParams ?? {};
+
+  const [selectedCase, setSelectedCase] = useState<string | null>(null);
+
+  const editSubActionProperty = useCallback(
+    (key: string, value: unknown) => {
+      const newProps = { ...actionParams.subActionParams, [key]: value };
+      editAction('subActionParams', newProps, index);
+    },
+    [actionParams.subActionParams, editAction, index]
+  );
+
+  const onCaseChanged = useCallback(
+    (id: string) => {
+      setSelectedCase(id);
+      editSubActionProperty('caseId', id);
+    },
+    [editSubActionProperty]
+  );
+
+  useEffect(() => {
+    if (!actionParams.subAction) {
+      editAction('subAction', 'addComment', index);
+    }
+
+    if (!actionParams.subActionParams?.caseId) {
+      editSubActionProperty('caseId', caseId);
+    }
+
+    if (!actionParams.subActionParams?.comment) {
+      editSubActionProperty('comment', comment);
+    }
+
+    // We excluded the caseId from the dependency array to avoid an infinity loop
+    if (caseId != null) {
+      setSelectedCase(caseId);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
why using `eslint-disable-next-line react-hooks/exhaustive-deps`

Reply:
Because it creates an infinitive loop if you add the `caseId` in the dependency array. You want to update the case once at the beginning.

Reply:
ok good to know, can we add a comment for `Future Engineer`?

Reply:
I think we can do that to avoid this problem
```
useEffect(() => {
    if (!actionParams.subAction) {
      editAction('subAction', 'addComment', index);
    }

    if (!actionParams.subActionParams?.caseId) {
      editSubActionProperty('caseId', caseId);
    }

    if (!actionParams.subActionParams?.comment) {
      editSubActionProperty('comment', comment);
    }

    if (caseId != null) {
      setSelectedCase((prevCaseId) => (prevCaseId !== caseId ? caseId : prevCaseId));
    }
  }, [
    actionConnector,
    index,
    actionParams.subActionParams,
    caseId,
    comment,
    actionParams.subAction,
    editAction,
    editSubActionProperty,
  ]);
```

Reply:
ok let's update the message to that now